### PR TITLE
Filter comments section

### PR DESCRIPTION
**Pull request by lumburovskalina**
Implements:
- https://github.com/code-kern-ai/refinery/issues/136

PRs:
- https://github.com/code-kern-ai/refinery-ui/pull/90

### CODE REVIEW

File: src/app/data/components/data-browser/data-browser.component.ts [hunk 1]
```diff
@@ -544,6 +550,17 @@ export class DataBrowserComponent implements OnInit, OnDestroy {
     return false;
   }
 
+  _commentsCreateSeachGroup(item: SearchGroupItem): FormGroup {
```
Comment (JWittmeyer) [inline src/app/data/components/data-browser/data-browser.component.ts]: shouldn'T this function be in the commentsFilter class?

File: src/app/data/components/data-browser/data-browser.component.ts [hunk 2]
```diff
@@ -1127,6 +1144,18 @@ export class DataBrowserComponent implements OnInit, OnDestroy {
...
+    if (!currentRecordIds || currentRecordIds.length == 0) return;
+
+    this.projectApolloService.getRecordComments(this.projectId, currentRecordIds).pipe(first()).subscribe((comments) => {
+      if (!comments) return;
+      this.recordComments = JSON.parse(comments).reduce(function (r, a) {
+        r[a.record_id] = r[a.record_id] || [];
+        r[a.record_id].push(a);
+        return r;
+      }, Object.create(null));
+    });
```
Comment (JWittmeyer) [inline src/app/data/components/data-browser/data-browser.component.ts]: i'd put this request in the comment class as well so here we would only see something like 

this.commentsFilter.collectRecordComments()

or something similar.

Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: also I'm not super sure about the logic of the function. ill take a close look if I can pinpoint what my issues are :D

File: src/app/data/components/data-browser/data-browser.component.ts [hunk 3]
```diff
@@ -1661,6 +1692,13 @@ export class DataBrowserComponent implements OnInit, OnDestroy {
     this.updateSearchParameters.refreshSearchParams(drillDown.getRawValue());
   }
 
+  processCommentsFilter(key: string, filterGroups: Object): void {
+    if (!this.fullSearch.has(key)) { this.displayOutdatedWarning = true; return; }
+    const comments: FormGroup = this.fullSearch.get(key) as FormGroup;
+    this.applyValuesToFormGroup(filterGroups[key], comments);
+    this.updateSearchParameters.refreshSearchParams(comments.getRawValue());
+  }
+
```
Comment (JWittmeyer) [inline src/app/data/components/data-browser/data-browser.component.ts]: is this now in both classes? (there is a processCommentsfilter in the class as well)

File: src/app/data/components/data-browser/helper-classes/filter-parser.ts [hunk 4]
```diff
@@ -294,6 +297,17 @@ export class DataBrowserFilterParser {
             appendTo.push(JSON.stringify(blacklist));
     }
 
+    private appendBlackAndWhiteListComments(appendTo: string[], searchElement: any): any {
+        let element = {
```
Comment (JWittmeyer) [inline src/app/data/components/data-browser/helper-classes/filter-parser.ts:302]: might be mistaken but don'T we need a check for active here somewhere?

**Comment by JWittmeyer** [comment]
databrowser now has a scrollbar on x (most likely because of the slightly to large tooltip on download records)
image

- [x] resolved

**Comment by JWittmeyer** [comment]
comment modal display options:
Comment XXX -> bold
max width of modal ~40 rem  
comment content -> word-break: break-word; (tailwind class break-words)
image

- [x] resolved

File: src/app/data/components/data-browser/data-browser.component.ts (hunk 2 above)
Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: found it, it simply doesn'T work :D 
test case: comment on record id 0 & on record id > 20 (so scrolling down will collect another batch)

Only the most recent comment is available so first comment from record id 0 then after scrolling only from id > 20

Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: i'll look for a fix

Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: This worked great for me

```suggestion
    const currentRecordIds = parsedRecordData?.map((record) => record.id);
    if (!currentRecordIds || currentRecordIds.length == 0) return;

    this.projectApolloService.getRecordComments(this.projectId, currentRecordIds).pipe(first()).subscribe((comments) => {
           if (!comments) return;
          const commentsParsed = tryParseJSON(comments);
          if (!commentsParsed) return;
          commentsParsed.forEach(e => {
            if (!this.recordComments[e.record_id]) this.recordComments[e.record_id] = [];
            this.recordComments[e.record_id].push(e);
          });
    });
```

Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: or not! sec still an error here :D

**Comment by JWittmeyer** [comment]
name doesn't have cursor pointer but is clickable
image

- [x] resovled

File: src/app/data/components/data-browser/data-browser.component.ts (hunk 1 above)
Comment (lumburovskalina) [reply src/app/data/components/data-browser/data-browser.component.ts]: Forgot to delete it from the data browser component.

File: src/app/data/components/data-browser/data-browser.component.ts (hunk 2 above)
Comment (JWittmeyer) [reply src/app/data/components/data-browser/data-browser.component.ts]: missed to clear the dict if we don't extend :) 

``` Javascript
if (extend) {
      this.extendedRecords.recordList = [
        ...this.extendedRecords.recordList,
        ...parsedRecordData,
      ];
    } else {
      this.extendedRecords.recordList = parsedRecordData;
      this.recordComments = {};  //<-------------------------------------- this one
    }

```

File: src/app/data/components/data-browser/helper-classes/filter-parser.ts (hunk 4 above)
Comment (lumburovskalina) [reply src/app/data/components/data-browser/helper-classes/filter-parser.ts:302]: Yes, I added the condition for active

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**